packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActionButton,
+  DialogActions,
+  DialogCancelButton,
+  DialogContent,
+  DialogHeader,
+  Form,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import { useState } from "react";
+
+import {
+  addUIDField,
+  getFieldLabel,
+  getUIDFieldEntry,
+  updateUIDField,
+} from "@/domain/customType";
+import { UIDFieldCustomErrorMap, UIDFieldLabelSchema } from "@/domain/fields";
+import { useCustomTypeState } from "@/features/customTypes/customTypesBuilder/CustomTypeProvider";
+
+export function UIDEditor() {
+  const [isOpen, setOpen] = useState(false);
+  const { customType, setCustomType } = useCustomTypeState();
+  const [_, field] = getUIDFieldEntry(customType) ?? [];
+  const uidFieldLabel = field ? getFieldLabel(field) : "";
+  const [label, setLabel] = useState(uidFieldLabel ?? "");
+  const [error, setError] = useState<string | undefined>();
+
+  function handleOpenChange(open: boolean) {
+    setOpen(open);
+    if (!open) {
+      setLabel(uidFieldLabel ?? "");
+      setError(undefined);
+    }
+  }
+
+  function handleValueChange(value: string) {
+    setLabel(value);
+    const result = UIDFieldLabelSchema.safeParse(value, {
+      errorMap: UIDFieldCustomErrorMap,
+    });
+    if (result.error) {
+      setError(result.error.errors[0].message);
+    } else {
+      setError(undefined);
+    }
+  }
+
+  function handleSubmit() {
+    if (Boolean(error)) {
+      return;
+    }
+    const updatedCustomType = field
+      ? updateUIDField(label, customType)
+      : addUIDField(label, customType);
+
+    setCustomType(updatedCustomType);
+    setOpen(false);
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={handleOpenChange}
+      size="small"
+      trigger={
+        <Button
+          color="grey"
+          textColor="placeholder"
+          startIcon={field ? "language" : "add"}
+          sx={{ marginInline: "auto" }}
+        >
+          {field ? uidFieldLabel : "Add an UID"}
+        </Button>
+      }
+    >
+      <DialogHeader title="Edit the UID label" />
+      <DialogContent>
+        <Form onSubmit={handleSubmit}>
+          <Box flexDirection="column" padding={16} gap={4}>
+            <FormInput
+              type="text"
+              label="Label *"
+              placeholder="UID"
+              value={label}
+              onValueChange={handleValueChange}
+              error={error}
+            />
+            {/* TODO: refactor if change proposed in: https://github.com/prismicio/editor/pull/1151 is released */}
+            {typeof error === "undefined" && (
```

Comment:
💡 #idea

I think you have a problem with your config, we never did that kind of things to check for undefined or not:

<img width="390" alt="Screenshot 2024-08-08 at 12 35 34" src="https://github.com/user-attachments/assets/b9dfbd6e-70e4-4842-9aea-9fa34b997586">

Reply:
ok, will need to check that, you think I should ignore those warnings for now?

Reply:
I think you should make sure VSCode is not telling you something different than us, could be hard for you to work :/